src/scripts/formatter.lua

```diff
@@ -21,25 +21,88 @@ local function build_sprite(class, name)
   return "[img="..class.."/"..name.."]"
 end
 
-local function get_properties(obj_data, force_index)
-  local researched
+local function get_properties(obj_data, force_index, options)
+  local opt = options or {}
+  local researched = false
   if obj_data.researched_forces then
     researched = obj_data.researched_forces[force_index] or false
-  else
-    researched = obj_data.available_to_all_forces or obj_data.available_to_forces[force_index] or false
+  elseif obj_data.available_to_all_forces or not obj_data.unlocked_by or #obj_data.unlocked_by == 0 or (obj_data.available_to_forces[force_index] and ((obj_data.temperatures_count and obj_data.temperatures_count <= 1) or not opt.fluid_temperature_key)) then
```

Comment:
I am not a fan of the huge amount of complexity required in this function. This function runs `thousands` of times and is extremely performance-sensitive. Instead of doing all of this temperature checking logic here, do it in the `on_research_finished` event and update a lookup table of which temperatures are available, that this function can simply check against.